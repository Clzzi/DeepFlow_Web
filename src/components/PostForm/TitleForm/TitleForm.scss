.TitleForm {
  display: flex;
  display: -webkit-flex;
  justify-content: space-between;
  align-items: flex-end;
  margin-bottom: 1rem;

  & > * {
    width: 50%;
    font-size: 2.75rem;
  }

  &-Input {
    outline: none;
    background: none;
    border: none;
    margin-right: 2rem;
    font-weight: bold;
    color: var(--contrast);

    &::placeholder {
      color: var(--lighterBlack);
    }
  }

  &-Title {
    max-width: 100%;
    overflow: hidden;
    text-overflow: ellipsis;
    outline: none;
    border: none;
    background: none;
    font-weight: bold;
  }
}