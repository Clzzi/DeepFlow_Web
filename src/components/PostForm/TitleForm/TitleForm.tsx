import { ChangeEvent } from 'react';
import classNames from 'classnames';
import { ClassNamesFn } from 'classnames/types';

const style = require('./TitleForm.scss');
const cx: ClassNamesFn = classNames.bind(style);

interface TitleFormProps {
  title: string;
  onChangeTitle: (e: ChangeEvent<HTMLInputElement>) => void;
}

const TitleForm = ({ title, onChangeTitle }: TitleFormProps): JSX.Element => {
  return (
    <div className={cx('TitleForm')}>
      <input
        type='text'
        className={cx('TitleForm-Input')}
        value={title}
        onChange={onChangeTitle}
        placeholder='제목을 입력하세요'
      />
    </div>
  );
};

export default TitleForm;
