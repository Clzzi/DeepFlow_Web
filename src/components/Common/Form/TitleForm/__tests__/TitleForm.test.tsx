import { ChangeEvent } from 'react';
import { fireEvent, render } from '@testing-library/react';
import TitleForm, { TitleFormProps } from '../TitleForm';

describe('TitleForm', () => {
  const initialSetup = (props: Partial<TitleFormProps> = {}) => {
    const initialProps: TitleFormProps = {
      title: '',
      onChangeTitle: (e: ChangeEvent<HTMLTextAreaElement>) => {},
      ...props,
    };

    return render(<TitleForm {...initialProps} />)
  };

  it('render correctly', () => {
    const { container } = initialSetup();
    expect(container).toMatchSnapshot();
  });

  it('props render correctly', () => {
    const wrapper = initialSetup({ title: '제목' });
    expect(wrapper.getByPlaceholderText('제목을 입력하세요')).toMatchSnapshot();
  });

  it('well handling change', () => {
    const titleForm: HTMLElement | null = document.getElementById('titleFormInput');
    if (titleForm !== null) {
      fireEvent.change(titleForm);
    }
  });
});