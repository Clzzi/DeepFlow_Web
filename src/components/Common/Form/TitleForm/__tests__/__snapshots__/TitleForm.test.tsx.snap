// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TitleForm props render correctly 1`] = `
<textarea
  class="TitleForm-Input"
  id="titleFormInput"
  name="title"
  placeholder="제목을 입력하세요"
  rows="1"
  style="overflow-x: hidden; word-wrap: break-word; overflow-y: hidden;"
>
  제목
</textarea>
`;

exports[`TitleForm render correctly 1`] = `
<div>
  <div
    class="TitleForm"
  >
    <textarea
      class="TitleForm-Input"
      id="titleFormInput"
      name="title"
      placeholder="제목을 입력하세요"
      rows="1"
      style="overflow-x: hidden; word-wrap: break-word; overflow-y: hidden;"
    />
    <textarea
      class="TitleForm-Input"
      readonly=""
      rows="1"
      style="overflow-x: hidden; word-wrap: break-word; overflow-y: hidden;"
      type="text"
    />
  </div>
</div>
`;
